server/tournament/match_stats_min.py

```diff
@@ -0,0 +1,171 @@
+        stats=match.stats,
+        team=team,
+        started_on=MatchEvent.Mode.OFFENSE,  # ignore this for now, not used
+        scored_by=scored_by,
+        assisted_by=assisted_by,
+        type=MatchEvent.EventType.SCORE,
+    )
+    new_match_event.save()
+
+    if match.team_1 is not None and match.team_2 is not None and team.id == match.team_1.id:
+        match.stats.score_team_1 += 1
+        match.stats.current_possession = match.team_2
+    if match.team_1 is not None and match.team_2 is not None and team.id == match.team_2.id:
+        match.stats.score_team_2 += 1
+        match.stats.current_possession = match.team_1
+
+    match.stats.save()
+
+    return 200, match.stats
+
+
+def handle_score_undo(
+    match: Match, score_event: MatchEvent
+) -> tuple[int, MatchStats | message_response]:
+    if match.team_1 is None or match.team_2 is None:
+        return 422, {"message": "Invalid teams"}
+
+    if score_event.team.id == match.team_1.id:
+        match.stats.score_team_1 -= 1
+        match.stats.current_possession = match.team_1
+
+    if score_event.team.id == match.team_2.id:
+        match.stats.score_team_2 -= 1
+        match.stats.current_possession = match.team_2
+
+    score_event.delete()
+    match.stats.save()
+
+    return 200, match.stats
+
+
+def handle_block(
+    match_event: MatchEventCreateSchema, match: Match, team: Team
+) -> tuple[int, MatchStats | dict[str, str]]:
+    if match_event.block_by_id is None:
+        return 422, {"message": "Block by is needed"}
+
+    try:
+        block_by = Player.objects.get(id=match_event.block_by_id)
+    except Player.DoesNotExist:
+        return 422, {"message": "Invalid Block by"}
+
+    if team.id == match.stats.current_possession.id:
+        return 422, {"message": "Team does not match the team in current defense"}
+
+    new_match_event = MatchEvent(
+        stats=match.stats,
+        team=team,
+        started_on=MatchEvent.Mode.OFFENSE,  # ignore this for now, not used
+        block_by=block_by,
+        type=MatchEvent.EventType.BLOCK,
+    )
+    new_match_event.save()
+
+    if match.team_1 is not None and match.team_1.id == team.id:
+        match.stats.current_possession = team
+    if match.team_2 is not None and match.team_2.id == team.id:
+        match.stats.current_possession = team
+
+    match.stats.save()
+
+    return 200, match.stats
+
+
+def handle_block_undo(
+    match: Match, block_event: MatchEvent
+) -> tuple[int, MatchStats | message_response]:
+    if match.team_1 is None or match.team_2 is None:
+        return 422, {"message": "Invalid teams"}
+
+    if block_event.team.id == match.team_1.id:
+        match.stats.current_possession = match.team_2
+
+    if block_event.team.id == match.team_2.id:
+        match.stats.current_possession = match.team_1
+
+    block_event.delete()
+    match.stats.save()
+
+    return 200, match.stats
+
+
+def handle_undo(match: Match) -> tuple[int, MatchStats | message_response]:
```

Comment:
you can put this on top and then followed by handle_score_undo and handle_block_undo so readablity be better